Class.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "Attribute.h"
+
+class Class {
+public:
+  Class();
+  Class(const std::string &name);
+  void rename(const std::string &newName);
+
+  const std::string &name() const;
+  const char *rawName() const;
+
+  void addField (const Field& field);
+  void addMethod (const Method& method);
+
+  void removeField (int index);
+  void removeField (const std::string& name);
+
+  void removeMethod (int index);
+  void removeMethod (const std::string& name);
+
+  Method& method (const std::string& name);
+  const Method& method (const std::string& name) const;
+  Method& method (int index);
+  const Method& method (int index) const;
+
+  Field& field (const std::string& name);
```

Comment:
As mentioned previously in the implementation files, you have a choice to make with method(..) and field(..)